### DISCUSSION TITLE
Building Rosbridge from Source -  No definition of [ros_babel_fish] for OS [nixos]

### DISCUSSION THREAD

**Discussion by camchambers** [foxglove/community#824] [General]
Hi, I'm trying to build foxglove_bridge from source according to the instructions [here](https://github.com/foxglove/ros-foxglove-bridge/#fetch-source-and-install-dependencies)  but I'm getting the following error:

```
$ rosdep install --ignore-src --default-yes --from-path src
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies: foxglove_bridge: No definition of [ros_babel_fish] for OS [nixos]
```

Using Ubuntu 20.04.5 + ROS noetic distro.

**Comment by achim-k** [comment]
I think you will be out of luck installing binary ROS packages with rosdep for your OS. It might work by overriding the OS with e.g. `rosdep install --os=ubuntu:focal` but I think you will run into other problems quite quickly (package manager etc). Your best bet is probably to clone [ros_babel_fish](https://github.com/StefanFabian/ros_babel_fish) (and other dependencies) into your workspace and build them together with foxglove_bridge.
